presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -1376,6 +1376,10 @@ private boolean containsAllPrivilege(Set<PrivilegeGrantInfo> requestedPrivileges
                 chosenAlternative.updateAndGet(currentChosen -> Math.min(currentChosen, position));
                 return result;
             }
+            catch (NoSuchObjectException e) {
+                // This is likely a valid response. We are not setting on an alternative yet, we will do it later when we get a more obviously valid response.
```

Comment:
setting -> settling
```java
// This is likely a valid response. We are not settling on an alternative yet,
// but will do it later when we get a more obviously valid response.

Reply:
Let it be. Although https://dictionary.cambridge.org/dictionary/english/settle-on-upon-sth suggests I could use "upon" here too 😉